.github/workflows/main.yml

```diff
@@ -1,82 +1,63 @@
-name: CI
-
+name: "Tests"
 on:
   push:
-    branches:    
-      - master         # Push events on master branch
-  pull_request:        # Run tests for any PRs
-
-env:
-  IMAGE_NAME: cfn-lint-action
-
+    branches:
+      - main
+      - ft/2.0.0
+  pull_request:
 jobs:
-  # Run tests.
-  test:
+  unit:
+    name: npm test
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v2
-
-      - name: Docker build
-        run: docker build . --file Dockerfile --tag image
-
-      - name: Docker run
-        run: docker run --entrypoint cfn-lint image -v
-
-  lint:
-    # Name the Job
-    name: Lint Code Base
-    # Set the agent to run on
-    runs-on: ubuntu-latest
-
-    ##################
-    # Load all steps #
-    ##################
+      - uses: actions/setup-node@v2
+      - run: npm ci
+      - run: npm test
+  integ:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-18.04
+          - ubuntu-20.04
+          - windows-2016
+          - windows-2019
+          - macos-10.15
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     steps:
-      ##########################
-      # Checkout the code base #
-      ##########################
-      - name: Checkout Code
+        # Step 1
+      - name: Git Checkout
         uses: actions/checkout@v2
-
-      ################################
-      # Run Linter against code base #
-      ################################
-      - name: Lint Code Base
-        uses: docker://github/super-linter:v2.2.0
-        env:
-          VALIDATE_ALL_CODEBASE: true
-
-  # Push image to GitHub Packages.
-  # See also https://docs.docker.com/docker-hub/builds/
-  push:
-    # Ensure test job passes before pushing image.
-    needs: test
-
-    runs-on: ubuntu-latest
-    if: github.event_name == 'push'
-
-    steps:
-      - uses: actions/checkout@v2
-
-      - name: Build image
-        run: docker build . --file Dockerfile --tag $IMAGE_NAME
-
-      - name: Log into GitHub Container Registry
-        run: echo "${{ secrets.CFN_CR_PAT }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
-
-      - name: Push image to GitHub Container Registry
+        # Step 2
+      - name: Setup Cloud Formation Linter with Latest Version
+        uses: ./
+        # Step 3
+      - name: Print the Cloud Formation Linter Version & run Linter.
+        run: |
+          cfn-lint --version
+          cfn-lint -t ./examples/template.yml
+        # Step 4
+      - name: Setup Cloud Formation Linter with Specific Version
+        uses: ./
+        with:
+          version: "0.44.5"
```

Comment:
This is the version of the CFN-LINT package someone may want to download from PyPi. One thing we had a use case for is to use a specific version of the cfn-lint library 👍 A consumer of this module can go to [here](https://pypi.org/project/cfn-lint/#history) and specify any version 👍 This allows users to be specific. Obviously, if they don't specify anything we default to the latest version 👍 

A future release idea is to make sure the user specifies a valid version. I try and do that with the regex but this can likey be made better :)